File: src/patterns/ignores.ts
import type { InitState } from "./initState.js"
import type { Resource } from "./resource.js"
import type { RuleMatch } from "./rule.js"

/**
 * Used in {@link IgnoresCb}.
 *
 * @since 0.6.0
 */
export interface IgnoresOptions extends InitState {
	/**
	 * File or directory without the slash suffix.
	 *
	 * @since 0.6.0
	 */
	entry: string
	/**
	 * Pre-lowercased entry path.
	 *
	 * @since 0.11.1
	 */
	lowerEntry?: string
	/**
	 * The associated resource.
	 *
	 * @since 0.11.0
	 */
	resource: Resource
	/**
	 * Result of the `dirname(entry)` call.
	 *
	 * @since 0.10.1
	 */
	parentPath: string
}

/**
 * @see {@link IgnoresCb}
 *
 * @since 0.11.0
 */
export type IgnoresCb = (
	options: IgnoresOptions,
	cb: (err: Error | null, match: RuleMatch) => void,
) => void

File: internal/shared/ignores.go
package shared

/**
 * Used in {@link Ignores}.
 *
 * @since 0.6.0
 */
type IgnoresOptions struct {
	InitState
	// File or directory without the slash suffix.
	//
	// # Since 0.6.0
	Entry string
}

// Checks whether a given entry path should be ignored based on its patterns.
//
// See [ResolveSources].
// See [SignedPattern.Ignores].
// See https://github.com/Mopsgamer/view-ignored/tree/main/src/targets for usage examples.
//
// # Since 0.6.0
type Ignores = func(options IgnoresOptions) SignedPatternMatch
